src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.cs

```diff
@@ -702,7 +715,16 @@ private void DocumentLoadCompleted(IVsTextBuffer shimTextBuffer, DocumentKey doc
                 throw new InvalidOperationException("The IVsTextBuffer has been populated but the underlying ITextBuffer does not exist!");
             }
 
-            NewBufferOpened(docCookie, textBuffer, documentKey, IsCurrentContext(docCookie, documentKey));
+            if (documentKey == null)
+            {
+                // Non-Roslyn document.
+                this.DocumentTrackingService.OnNonRoslynBufferOpened(textBuffer);
+                _docCookiesToNonRoslynDocumentBuffers.Add(docCookie, textBuffer);
```

Comment:
Isn't this duplicated with the "if already open" logic up around line 390? Can we refactor?